src/handlers/mod.rs

```diff
@@ -425,3 +428,56 @@ impl DrmLeaseHandler for State {
     }
 }
 delegate_drm_lease!(State);
+
+impl GammaControlHandler for State {
+    fn gamma_control_manager_state(&mut self) -> &mut GammaControlManagerState {
+        &mut self.niri.gamma_control_manager_state
+    }
+
+    fn set_gamma(&mut self, wl_output: &WlOutput, ramp: Vec<u16>) -> Result<(), ()> {
+        if !self.backend.is_tty() {
+            error!("Backend is not Tty");
+            return Err(());
+        }
+
+        let Some(output) = Output::from_resource(wl_output) else {
+            return Err(());
+        };
+
+        self.backend.tty().set_gamma(&output, ramp);
+        Ok(())
+    }
+
+    fn get_gamma(&mut self, wl_output: &WlOutput) -> Option<Vec<u16>> {
+        if !self.backend.is_tty() {
+            error!("Backend is not Tty");
+            return None;
+        }
+
+        let Some(output) = Output::from_resource(wl_output) else {
+            return None;
+        };
+
+        self.backend.tty().get_gamma(&output)
+    }
+
+    fn destroy(&mut self, output_id: ObjectId) {
+        self.niri
+            .gamma_control_manager_state
+            .destroy_gamma_control(output_id)
+    }
```

Comment:
Is this really needed if it just calls a method on the gamma_control_manager_state itself?

Reply:
I think it is, because we go from one impl to another through the state. But I might be wrong...